### PR TITLE
Override browser timing functions

### PR DESCRIPTION
**Pull request by starkbamse**
The basic issue is as follows - setTimeout and setInterval have one major problem in the browser environment - if the tab goes away in the background, eventually Chrome V8 (JS) stops firing them. This can be desired or an undesired behavior, but it is a reality.  This breaks processing logic in things like interval.

The only way around this, is to start a worker task in chrome as workers don't go to sleep, then have it post messages to the main page.

So this PR injects custom setTimeout, setInterval, clearTimeout, clearInterval functions into the global window object and using workers ensures these never go out of focus.

### CODE REVIEW

File: core/src/wasm/overrides.rs [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+        js_sys::Reflect::set(&request, &JsValue::from("id"), &JsValue::from(id)).unwrap();
+
+        // Post the message to the worker
+        self.worker.post_message(&request).unwrap();
+
+        // Delete the callback from the map
+        self.callbacks.delete(&JsValue::from(id));
+    }
+
+    #[wasm_bindgen(js_name = "setTimeout")]
```
Comment (aspect) [inline core/src/wasm/overrides.rs:170]: As I understand there is no need to export `setTimeout` and other functions

File: core/src/wasm/overrides.rs [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    // Wrap the clearTimeout function
+    let clear_timeout_closure = {
+        let timer_manager = timer_manager.clone();
+        Closure::wrap(Box::new(move |id: u32| {
+            timer_manager.borrow_mut().clear_timeout(id);
+        }) as Box<dyn FnMut(u32)>)
+    };
+
+    // Set the new functions on the window object
+    js_sys::Reflect::set(
```
Comment (aspect) [inline core/src/wasm/overrides.rs:308]: pair these with declarations - make closure, override, make closure, override...  

(as opposed to make 4 closures and override all of them)...

makes code grouped better (IMO)

File: core/src/wasm/interval.rs [hunk 3]
```diff
@@ -67,6 +69,9 @@ pub struct Interval {
 impl Interval {
     /// Create a new `Interval` stream that will resolve each given duration.
     pub fn new(period: Duration) -> Self {
+        if let Err(e) = init_timer_overrides() {
+            workflow_log::log_error!("Unable to initialize timer overrides: {:?}", e);
```
Comment (aspect) [inline core/src/wasm/interval.rs:73]: Since you changed return type to String, no longer to output using debug... in fact you can just `log_error!(e);`

**Comment by starkbamse** [comment]
@aspect

File: core/src/wasm/interval.rs (hunk 3 above)
Comment (starkbamse) [reply core/src/wasm/interval.rs]: a2f25c0efe6a1120867a044ce3e95715ed5d9804

File: core/src/wasm/overrides.rs (hunk 2 above)
Comment (starkbamse) [reply core/src/wasm/overrides.rs]: a2f25c0efe6a1120867a044ce3e95715ed5d9804

File: core/src/wasm/overrides.rs (hunk 1 above)
Comment (starkbamse) [reply core/src/wasm/overrides.rs]: a2f25c0efe6a1120867a044ce3e95715ed5d9804

**Pull request merged**